File: Ruby/9 - Recursividad.rb
def factorial(n)
    if n == 0 || n == 1
        return 1
    else
        return n * factorial(n - 1)
    end 
end

def fibonacci(n)
    if n == 1 || n == 2
        return 1
    else
        return fibonacci(n - 1) + fibonacci(n - 2)
    end 
end


if __FILE__ == $0
    for i in 0..11
        puts "#{i}! = #{factorial(i)}"
    end 

    for i in 1..11
        puts "fibonacci_{#{i}} = #{fibonacci(i)}"
    end 
end 

File: R/9 - Recursividad.R
factorial <- function(n) {
  if (n == 0 || n == 1) {
    return(1)
  } else {
    return(n * factorial(n - 1))
  }
}

fibonacci <- function(n) {
  if (n == 1 || n == 2) {
    return(1)
  } else {
    return(fibonacci(n - 1) + fibonacci(n - 2))
  }
}

for (i in 0 : 10) {
  cat(paste(i, "! =", factorial(i), "\n"))
}

for (i in 1 : 30) {
  cat(paste("fibonacci_", i, "=", fibonacci(i), "\n"))
}
